test/src/code_field/code_controller_outdent_selection_test.dart

```diff
@@ -0,0 +1,406 @@
+          controller.text,
+          reason: example.name,
+        );
+        expect(
+          controller.value.selection,
+          example.expectedSelection,
+          reason: example.name,
+        );
+      }
+    });
+
+    test('Selection is a range', () {
+      final examples = [
+        _Example(
+          name: 'WHEN the entire document is selectd '
+              'SHOULD outdent all lines',
+          initialFullText: '''
+  AAAA
+      AAAA
+  AAAA''',
+          expectedFullText: '''
+AAAA
+    AAAA
+AAAA''',
+          expectedVisibleText: '''
+AAAA
+    AAAA
+AAAA''',
+          initialSelection: TextSelection(
+            baseOffset: 0,
+            extentOffset: 24,
+          ),
+          expectedSelection: TextSelection(
+            baseOffset: 0,
+            extentOffset: 18,
+          ),
+        ),
+        _Example(
+          name: 'WHEN lines that doesn\'t have indent are selected '
+              'SHOULD NOT outdent that lines',
+          initialFullText: '''
+AAAA
+      AAAA
+  AAAA
+''',
+          expectedFullText: '''
+AAAA
+    AAAA
+AAAA
+''',
+          expectedVisibleText: '''
+AAAA
+    AAAA
+AAAA
+''',
+          initialSelection: TextSelection(
+            baseOffset: 0,
+            extentOffset: 22,
+          ),
+          expectedSelection: TextSelection(
+            baseOffset: 0,
+            extentOffset: 19,
+          ),
+        ),
+        _Example(
+          name: 'Outdent SHOULD NOT unfold folded comment at line 0 '
+              'and folded imports',
+          initialFullText: '''
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.examples;
+
+import java.util.Arrays;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Filter;
+import org.apache.beam.sdk.transforms.FlatMapElements;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.TypeDescriptors;
+
+  aaaa {
+
+  }
```

Comment:
```suggestion
  a;
```
It's hard to see whether there are any spaces on the line between `{` and `}`. Make all tests as short as is meaningful because it's hard to maintain them.